python_modules/dagster/dagster/_core/workspace/workspace.py

```diff
@@ -40,29 +41,17 @@ class CodeLocationStatusEntry(NamedTuple):
     update_timestamp: float
 
 
-class IWorkspace(ABC):
-    """Manages a set of CodeLocations."""
-
-    @abstractmethod
-    def get_code_location(self, location_name: str) -> "CodeLocation":
-        """Return the CodeLocation for the given location name, or raise an error if there is an error loading it."""
-
-    @abstractmethod
-    def get_code_location_entries(self) -> Mapping[str, CodeLocationEntry]:
-        """Return an entry for each location in the workspace."""
-
-    @abstractmethod
-    def get_code_location_statuses(self) -> Sequence[CodeLocationStatusEntry]:
-        pass
+@dataclass
+class CodeLocationEntriesSnapshot:
+    code_location_entries: Mapping[str, CodeLocationEntry]
```

Comment:
*  seeing [1] wondering why this isnt called `WorkspaceSnapshot`, alternatively rename those args from `workspace_snapshot` to `code_location_entries_snapshot` / `code_locs_snapshot` or smth
* should be able to use `@record` tho doesn't matter much